src/app/utils/url-generator.js

```diff
@@ -9,19 +9,35 @@ export const host = () => {
 
 export const protocolAndHost = () => `${location.protocol}//${host()}`;
 
+export const newSandboxUrl = () => `/s/new`;
+
+const sandboxGitUrl = (git: {
+  repo: string,
+  branch: string,
+  username: string,
+  path: string,
+}) =>
+  'github/' + // eslint-disable-line prefer-template
+  encodeURIComponent(git.username) +
+  '/' +
+  encodeURIComponent(git.repo) +
+  '/tree/' +
+  encodeURIComponent(git.branch) +
+  '/' +
+  encodeURIComponent(git.path);
```

Comment:
I have a cleaner version of `sandboxGitUrl` running. It will need to be converted to TS.
```js
const sandboxGitUrl = ({ username, repo, branch, path }) => (
  buildEncodedUri`/s/github/${username}/${repo}/tree/${branch}/` + path
);
```

It uses a [Tagged Template Literal](https://www.styled-components.com/docs/advanced#tagged-template-literals) (the same thing used in `styled-components`) that I wrote called `buildEncodedUri` to call `encodeURIComponent` with each value in the template.

```js
const buildEncodedUri = (strings, ...values) => (
  strings[0] + values.map((value, i) => (
    `${encodeURIComponent(value)}${strings[i + 1]}`
  )).join('')
);
```

It might be little overkill for this, but it's reusable for other URI building applications.

see it here on [codesandbox](https://codesandbox.io/s/RoNWKDJqK) (you may have heard of it :)